### ISSUE TITLE
Toggle for screenshot detection API introduced with Android 14

### ISSUE THREAD

**Issue by mtkmgh** [GrapheneOS/os-issue-tracker#2590]
Somewhat related to [#664](https://github.com/GrapheneOS/os-issue-tracker/issues/664) and [#1915](https://github.com/GrapheneOS/os-issue-tracker/issues/1915). 

Similar to the issues mentioned above it would be great to get the ability to disable Android 14's screenshot detection API (either per app or globally) to prevent certain apps from being able to tell when a screenshot was taken. As there still are ways to prevent this screenshot detection by using the ADB or other tools, this would be a convenience feature, sparing the need for cumbersome workarounds.

**Comment by alexmo1997** [comment]
This is now more important with Android 15, as apps can now detect being screenshared, so it isn't easy to circumvent this anymore.

**Comment by stevenroose** [comment]
Isn't this a duplicate of #664?

**Comment by ChillyRelleno** [comment]
Just want to chime in in favor! Screenshot detection feels very invasive and makes me uncomfortable. 

Appreciate the work everyone does!

**Comment by DetachHead** [comment]
I don't think this should be marked as low priority IMO. I've been very happy with graphene OS since switching to it a few months ago, but this is an extremely privacy invading feature that fell through the cracks

**Comment by thestinger** [comment]
You're welcome to work on it if you think it's important. Posting comments requesting a feature won't result in a higher priority or faster progress. The thumbs up reaction on the initial issue should be used for express support for a feature.

**Comment by solitudery** [comment]
How would this work with messaging apps that don't allow screenshotting a private chat, would disabling screenshot detection bypass that?

**Comment by thestinger** [comment]
@solitudery That's a separate feature we also plan to implement.

**Comment by DetachHead** [comment]
i understand the privacy concerns regarding screenshotting private chats, but IMO it should not be up to an app to decide whether or not i can take a screenshot. that just sounds like the same thing as the DRM in streaming services that prevent screenshots and screen recording.

**Comment by thestinger** [comment]
That's off topic for this issue. We have https://github.com/GrapheneOS/os-issue-tracker/issues/664 for that.

**Comment by Meijuta** [comment]
I am concerned this may prompt certain apps to block grapheneOS (possibly via PIA?). 
So I think its best to have a toggle for ss detection, where the default setting is for it to be enabled. Hopefully preventing inflammatory reactions.

**Comment by 0xALUCARD** [comment]
Implemented in https://github.com/GrapheneOS/platform_frameworks_base/pull/447
